### ISSUE TITLE
sbt-sri-platform should generate folders "android" and "ios"

### ISSUE THREAD

**Issue by frgomes** [scalajs-react-interface/incubator#8]
At the moment, the developer has to generate folders for androi and ios like below:

    $ project=myAwesomeProject
    $ react-native init $project
    $ mv $project/android $project/ios .
    $ rm -r -f $project

This should be automated by our SBT plugin
https://github.com/scalajs-react-interface/sbt-sri-platform

Tasks:
 - [x] investigate requirements and limitations;
 - [ ] implement generation of folder for Android;
 - [ ] implement generation of folder for iOS;

**Comment by frgomes** [comment]
## Investigation

### Android
Apparently easy: g8 should be able to generate the entire folder.

However, my tests demonstrated that the folder is generated empty. The root cause is unknown.

### iOS
There are several hashes that need to be calculated, which cannot be done in g8.

Also, it seems like XCode is a moving target. Certain developers report that Apple keeps moving things around and breaking developer's code. So, from this perspective, we should not join the group of users affected by modifications in XCode.

I recommend that we keep the manual procedure in the case of iOS:

    $ project=myAwesomeProject
    $ react-native init $project
    $ mv $project/ios .
    $ rm -r -f $project

**Comment by frgomes** [comment]
@virtualirfan 
I think that ``sbt-sri-platform`` should provide comands for automating this sort of mundane tasks.
* Apparently, it's just a handful of ``bash`` commands in the case of Android; and I would say that these commands should ideally be fired automagically by SBT when necessary.
* The case of IOS seems more complicated and I'd say that it only makes sense to support IOS after we gain experience with such procedure for Android.

**Comment by virtualirfan** [comment]
Not sure why things are harder for `ios` versus `android`.

`$ react-native init $project` creates both `ios` and `android` projects and all that has to happen is to `mv` and `rm`, right? So, why not just have the sbt plugin just run the bash commands to generate files and then `mv/rm`?

I'm probably missing something.

I'm not in favor of two different process for `ios` and `android` :-(

**Comment by frgomes** [comment]
@virtualirfan : You are correct. The evaluation above was done long time ago and I was considering a different strategy. If we consider that SBT can run arbitrary commands (and it can!) we can automate the entire process which is now done by hand. From this perspective... yes: both Android and IOS are conceptually the same: just a handful of commands which prepare the environment.
